cmd/soroban-rpc/internal/db/ledger.go

```diff
@@ -23,20 +25,70 @@ type LedgerReader interface {
 	StreamAllLedgers(ctx context.Context, f StreamLedgerFn) error
```

Comment:
is `BatchGetLedgers()` ever called outside of `LedgerReaderTx`? If not then it seems like you can remove `LedgerReaderTx()` from the `LedgerReader` interface

Reply:
@tamirms Right now its not. I was going to remove it but thought that it could be used in other places in future without the need of a tx so kept it in `LedgerReader` too

Reply:
I think it's better to remove unused code and add it back when it becomes necessary